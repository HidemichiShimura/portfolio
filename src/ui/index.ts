// Atoms
export { default as PageHeading } from "./atoms/PageHeading/PageHeading";
export { default as Avatar } from "./atoms/Avatar/Avatar";
export { default as Typography } from "./atoms/Typography/Typography";
export { default as SectionHeading } from "./atoms/SectionHeading/SectionHeading";
export { default as Icon } from "./atoms/Icon/Icon";
export { default as RadialLinearContainer } from "./atoms/RadialLinearContainer/RadialLinearContainer";
export { default as Grid } from "./atoms/Grid/Grid";
export { default as GradientCard } from "./atoms/GradientCard/GradientCard";
export { default as Card } from "./atoms/Card/Card";

// Molecules
export { default as HeroHeadings } from "./molecules/HeroHeadings/HeroHeadings";
