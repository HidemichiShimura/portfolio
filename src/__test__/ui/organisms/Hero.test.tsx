import { Hero } from "@/ui";
import { screen, render } from "@testing-library/react";

const NAME = "Hidemichi Shimura";
const POSITION = "Web Developer";
const DESCRIPTION = "Challenge Seeker";

describe("Organisms: <Hero />", () => {
  test("renders", () => {
    render(<Hero />);

    expect(screen.getByTestId("hero")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    render(<Hero />);

    expect(screen.getByTestId("hero")).toMatchSnapshot();
  });

  test("displays my name", () => {
    render(<Hero />);

    expect(screen.queryByText(NAME)).toBeInTheDocument();
  });

  test("displays my positon", () => {
    render(<Hero />);

    expect(screen.queryByText(POSITION)).toBeInTheDocument();
  });

  test("displays my description", () => {
    render(<Hero />);

    expect(screen.queryByText(DESCRIPTION)).toBeInTheDocument();
  });
});
