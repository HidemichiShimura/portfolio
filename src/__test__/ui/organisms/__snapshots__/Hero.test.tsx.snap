// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms: <Hero /> matches snapshot 1`] = `
<div
  class="grow"
  data-testid="hero"
>
  <section
    class="bg-radial-gradient animate-radial-gradient bg-[radial-gradient(circle,#7952B3,transparent,transparent)] bg-[length:100px_100px] bg-no-repeat md:bg-[length:350px_350px]"
  >
    <div
      class="flex h-screen w-full items-center justify-center self-stretch"
    >
      <div
        class="flex flex-col items-center gap-8"
      >
        <div
          class="text-center"
        >
          <h1
            class="font-60s-stripe dark:text-neutral-100 text-light-gray text-6xl md:text-8xl"
          >
            Hidemichi Shimura
          </h1>
        </div>
        <div
          class="flex flex-col items-center"
        >
          <h2
            class="font-60s-stripe dark:text-neutral-100 text-light-gray text-2xl md:text-4xl"
          >
            Web Developer
          </h2>
          <h2
            class="font-60s-stripe dark:text-neutral-100 text-light-gray text-2xl md:text-4xl"
          >
            &
          </h2>
          <h2
            class="font-60s-stripe dark:text-neutral-100 text-light-gray text-2xl md:text-4xl"
          >
            Challenge Seeker
          </h2>
        </div>
      </div>
    </div>
  </section>
</div>
`;
